Formula/llvm.rb

```diff
@@ -381,8 +402,21 @@ def caveats
               "-I#{MacOS.sdk_path}/usr/include",
               "-L#{lib}",
               "-Wl,-rpath,#{lib}", "test.cpp", "-o", "test"
-      assert_includes MachO::Tools.dylibs("test"), "#{opt_lib}/libc++.1.dylib"
+      assert_includes MachO::Tools.dylibs("test"), "@rpath/libc++.1.dylib"
```

Comment:
@DomT4 see https://github.com/Homebrew/homebrew-core/pull/10808. This belongs in a totally separate PR.